### PR TITLE
[Event Hubs Client] Test Resources Clean-Up

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to perform some clean-up and tweaks to the live tests and their associated Infrastructure.  Included are:

  - Refactoring of the `TestEnvironment` and `EventHubScope` to simplify
    some of the logic and reduce levels of indirection.

  - Remove the test-specific `ConnectionStringTokenParser`, given that
    the standard `ConnectionStringParser` was required to be part of the
    shared code assets and can now be accessed by tests or explicitly Included
    in the test project.

  - Ensure that the dynamic resource creation infrastructure is ignoring the
    synchronization context for asynchronous operations.

  - Include a test resource creation template to integrate with the Engineering
    infrastructure to dynamically create/remove resource groups for test runs.

  - Refactoring the Consumer live tests to make use of common infrastructure for
    the "read until empty" loop that many were using.

  - Ensured that any `TaskCompletionSource` instances created are specifying
    `TaskCreationOptions.RunContinuationsAsynchronously` to avoid potential deadlocks.

# Last Upstream Rebase

Friday, February 21, 4:12pm (EST)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs [hunk 1]
```diff
@@ -37,27 +38,19 @@ public static class TestEnvironment
 
         /// <summary>The environment variable value for the namespace connection string, lazily evaluated.</summary>
         private static readonly Lazy<string> EventHubsNamespaceConnectionString =
-            new Lazy<string>(() => ReadConnectionStringFromEnvironment("EVENT_HUBS_OVERRIDE_NAMESPACE_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
+            new Lazy<string>(() =>  Environment.GetEnvironmentVariable("EVENT_HUBS_NAMESPACE_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs]: nit: extra blank space after `=>`.

File: sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs [hunk 2]
```diff
@@ -37,27 +38,19 @@ public static class TestEnvironment
...
 
         /// <summary>The environment variable value for the event hub name, lazily evaluated.</summary>
         private static readonly Lazy<string> EventHubsEventHubName =
-            new Lazy<string>(() => ReadEnvironmentVariable("EVENT_HUBS_OVERRIDE_EVENT_HUB_NAME"), LazyThreadSafetyMode.PublicationOnly);
+            new Lazy<string>(() =>  Environment.GetEnvironmentVariable("EVENT_HUBS_OVERRIDE_EVENT_HUB_NAME"), LazyThreadSafetyMode.PublicationOnly);
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs]: nit: extra blank space after `=>`.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageTestEnvironment.cs [hunk 3]
```diff
@@ -14,21 +14,21 @@ namespace Azure.Messaging.EventHubs.Processor.Tests
...
     {
-        /// <summary>The active Azure storage connection for this test run, lazily created.</summary>
-        private static readonly Lazy<StorageScope.StorageProperties> ActiveStorageAccount =
-            new Lazy<StorageScope.StorageProperties>(EnsureStorageAccount, LazyThreadSafetyMode.ExecutionAndPublication);
-
         /// <summary>The environment variable value for the storage account connection string, lazily evaluated.</summary>
-        private static readonly Lazy<string> ActiveStorageAccountConnectionString =
-            new Lazy<string>(() => ReadEnvironmentVariable("EVENT_HUBS_OVERRIDE_BLOBS_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
+        private static readonly Lazy<string> StorageAccountConnectionString =
+            new Lazy<string>(() =>  Environment.GetEnvironmentVariable("EVENT_PROCESSOR_STORAGE_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageTestEnvironment.cs]: nit: extra blank space after `=>`.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs [hunk 4]
```diff
@@ -438,54 +367,36 @@ await using (EventHubScope scope = await EventHubScope.CreateAsync(1))
 
                 await using (var connection = new EventHubConnection(connectionString))
                 {
-                    var partition = (await connection.GetPartitionIdsAsync(DefaultRetryPolicy)).First();
+                   var partition = (await connection.GetPartitionIdsAsync(DefaultRetryPolicy)).First();
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs]: nit: indentation is off.

**Review by kinelski — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageTestEnvironment.cs [hunk 5]
```diff
@@ -47,46 +47,62 @@ public static class StorageTestEnvironment
...
-        private static StorageScope.StorageProperties EnsureStorageAccount()
+        private static StorageProperties EnsureStorageAccount()
         {
-            if (!string.IsNullOrEmpty(ActiveStorageAccountConnectionString.Value))
+            if (!string.IsNullOrEmpty(StorageAccountConnectionString.Value))
             {
-                return StorageScope.PopulateStoragePropertiesFromConnectionString(ActiveStorageAccountConnectionString.Value);
+                var connectionString = StorageAccountConnectionString.Value;
+                var nameStart = (connectionString.IndexOf('=', connectionString.IndexOf("AccountName")) + 1);
+             	var nameLength = (connectionString.IndexOf(';', nameStart) - nameStart);
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageTestEnvironment.cs]: Oddly, not in the IDE....    reset the margin and reindented.

**Pull request merged**